### PR TITLE
Implement basic Group view page

### PR DESCRIPTION
**Pull request by T-Dnzt**
# Introduction

This pull request introduces the group view page showing all the information about a specific group as well as the related resources, lists and comments. 

It also includes the addition of Pundit to handle permissions for the various actions of the `GroupsController` and `MembersController`. Note that the permissions are currently shared and only defined in the `GroupPolicy` class. Inside the `MembersController`, all actions adding/changing/removing members are associated with updating a group (only admins can do it). The alternative would be to change the Pundit helper method to take three arguments instead of two (user, group_user, group). 

# Related issues

- [Issue #61](https://github.com/greencommons/commons/issues/61) 
- [Issue #75](https://github.com/greencommons/commons/issues/75)

# Screenshots

screen shot 2559-12-16 at 12 11 03 pm

screen shot 2559-12-23 at 8 52 21 am

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,7 @@ gem 'sass-rails', '~> 5.0'
 gem 'sidekiq'
 gem 'slim'
 gem 'uglifier', '>= 1.3.0'
+gem 'pundit'
```
Comment (adarsh) [inline Gemfile]: I approve of this gem 😉

File: app/controllers/members_controller.rb [hunk 2]
```diff
@@ -27,18 +28,33 @@ def create
...
 
+  def leave
+    @group_current_user = @group.find_member(current_user)
+    authorize @group_current_user
+
+    if !@group_current_user.admin? || @group.admin_count > 1
+      @group_current_user.destroy
+      redirect_to @group, notice: 'You are no longer a member of this group.'
+    else
+      redirect_to @group, notice: 'You cannot leave this group because you are the only admin.'
```
Comment (ptrikutam) [inline app/controllers/members_controller.rb]: Great catch :)

File: app/models/group.rb [hunk 3]
```diff
@@ -5,11 +5,12 @@ class Group < ApplicationRecord
...
   has_many :users, through: :groups_users
   has_many :lists, as: :owner
+  has_many :resources, through: :lists
 
   validates :name, presence: true
 
-  def resources
-    lists.map(&:resources).flatten.uniq
+  def latest_resources(limit: 4)
+    resources.sort_by_created_at.distinct.limit(limit)
```
Comment (ptrikutam) [inline app/models/group.rb]: I dig it. If I'm not mistaken, this change will move these operations to the database level, and not do them in memory, correct?

File: spec/policies/group_policy.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    it { is_expected.to forbid_action(:index)   }
+    it { is_expected.to forbid_action(:new)     }
+    it { is_expected.to forbid_action(:create)  }
+    it { is_expected.to forbid_action(:update)  }
+    it { is_expected.to forbid_action(:edit)    }
+    it { is_expected.to forbid_action(:destroy) }
+  end
+
+  context 'when regular group member' do
```
Comment (ptrikutam) [inline spec/policies/group_policy.rb]: Should there be another spec here for when a person is logged in, but not a member?

I would imagine the permitted actions are then: `:show`, `:index`, `:new`, `:create`.

Unless I'm confusing what those policies are meant for... What do you think?

File: app/views/groups/show.html.slim [hunk 5]
```diff
@@ -1,10 +1,142 @@
...
+        = render 'shared/components/btn_count', path: '#', 
+                                                text: 'ADD TO LIST', 
+                                                label: @group.lists.count,
+                                                classes: 'resource-actions__button--right',
+                                                style: 'primary'
+        
+        - if @group_current_user                                 
+          = link_to 'LEAVE GROUP', leave_group_members_path(@group), 
+                                   method: :delete, 
+                                   data: { confirm: 'Are you sure?' },
```
Comment (ptrikutam) [inline app/views/groups/show.html.slim]: Can we update this to "Are you sure you want to leave this group?"

File: app/models/group.rb (hunk 3 above)
Comment (T-Dnzt) [inline app/models/group.rb]: Correct, it will all happen in one SQL query.

File: spec/policies/group_policy.rb (hunk 4 above)
Comment (T-Dnzt) [inline spec/policies/group_policy.rb]: I think you're right, indeed ;) Let me look into it.

Comment (T-Dnzt) [inline spec/policies/group_policy.rb]: And thanks for pointing that out, made me realize I'm missing the `_spec` suffix in the filename...

File: app/controllers/members_controller.rb (hunk 2 above)
Comment (T-Dnzt) [inline app/controllers/members_controller.rb]: Without this, we can say hello to orphan groups :D

**Pull request merged**